nixos/modules/installer/cd-dvd/sd-image-aarch64.nix

```diff
@@ -45,8 +57,17 @@
       '';
       in ''
         (cd ${pkgs.raspberrypifw}/share/raspberrypi/boot && cp bootcode.bin fixup*.dat start*.elf $NIX_BUILD_TOP/firmware/)
-        cp ${pkgs.ubootRaspberryPi3_64bit}/u-boot.bin firmware/u-boot-rpi3.bin
+
+        # Add the config
         cp ${configTxt} firmware/config.txt
+
+        # Add pi3 specific files
+        cp ${pkgs.ubootRaspberryPi3_64bit}/u-boot.bin firmware/u-boot-rpi3.bin
+
+        # Add pi4 specific files
+        cp ${pkgs.ubootRaspberryPi4_64bit}/u-boot.bin firmware/u-boot-rpi4.bin
+        cp ${pkgs.raspberrypi-armstubs}/armstub8-gic.bin firmware/armstub8-gic.bin
+        cp ${pkgs.raspberrypifw}/share/raspberrypi/boot/bcm2711-rpi-4-b.dtb firmware/
```

Comment:
So why is this one not picked up by `/dtbs/broadcom` in linux-rpi.nix?

Reply:
Because otherwise for the default mainline image, which is not raspberry-pi specific, we would build *two whole kernels*! We lucked-out in the fact that (1) we already were using `raspberrypifw` and (2) the foundation ships "golden" FDTs for us.

An alternative solution would have been to make the kernel building infra of NixOS able to only build device trees, so it would still have included another kernel's source in the build closure, but would have only built the minimum required to build the FDT.